ProcessLib/Process.h

```diff
@@ -164,8 +164,11 @@ class Process
 		{
 			MeshLib::Location const l(_mesh.getID(),
 			                          MeshLib::MeshItemType::Node, i);
-			auto const global_index = std::abs(
+			auto global_index = std::abs(
 			    _local_to_global_index_map->getGlobalIndex(l, component_id));
+			// Ghost entry and its original index is 0
+			if ( global_index == _x->size() )
```

Comment:
I don't like it, that now this check is scattered all across. But that can be changed later on. Could you add a TODO for it somewhere, please?